### ISSUE TITLE
OnlyOffice files not saving and file changes lost upon exit

### ISSUE THREAD

**Issue by alexrojo77** [ONLYOFFICE/DesktopEditors#1436]
### This issue is unique.

- [X] I have used the [search tool](https://github.com/ONLYOFFICE/DesktopEditors/issues?q=) and did not find an issue describing my bug.

### Operating System

Linux (DEB package)

### Version information

7.5.1.23

### Expected Behavior

Opened docx and xlsx files stored on network NAS (Synology; DSM7.2) via Nautilus file browser on Ubuntu 22.04.  Made some changes then saved/closed file.  Of course, I expected my changes to be saved upon exit and upon re-opening files but when I reopened the files the changes I had made were not there.  Repeated this and observed same thing (ie changes were not saved).  This is the first time I've observed this happening and was definitely working on a previous version of OnlyOffice.

### Actual Behavior

Changes to docx and xlsx files are not being saved on exit.  There was no indication of any errors when saving the file or exiting the program.  OnlyOffice actually showed briefly in the window that it was saving the file.  On the Nautilus file browser, I could also see that the files had not been modified based on the date/time.

### Reproduction Steps

1.  Open Nautilus file browser on Ubuntu Linux.
2. Open mapped network drive to document location on Synology (location mapped via SMB or DAVS share).
3. Double click on document or spreadsheet file.
4. Make changes as necessary.
5. Save file (either through save icon or menu or ctrl-s).
6. Close program.
7. Re-Open file.  [file contents unchanged]

### Additional information

I opened the same files using LibreOffice and the changes were properly saved.  Seems like the issue is with OnlyOffice so I uninstalled this version of OnlyOffice and installed a previous release of OnlyOffice from GitHub (7.5.0).  I was able to open the same files, make changes and save them properly.

**Comment by cezarq** [comment]
Same issue here running Ubuntu and OnlyOffice version 7.5.1.23 (snap)

**Comment by dernippel** [comment]
I can confirm this issue using Linux Mint 21.2 and the Flatpak package version 7.5.1.

**Comment by kabe-inc** [comment]
Same here, with both the latest .deb and flatpak versions.

Saving changes to a .docx or .xlsx file works if saving to a local folder, but the changes are lost if I save directly to my OneDrive cloud folder.

**Comment by doryiii** [comment]
Can confirm this is reproducible consistenly on 8.0.1:
- Mount a network drive using GVFS with Nautilus (e.g. "sftp://some-server")
- Open an existing .docx file or .xlsx file with Onlyoffice
- Made some changes
- Ctrl-S to save. Save toast notification popped up saying "saving document" and went away, just like normal saving.
- Close Onlyoffice editor. 
- Reopen the file
- Observe old content instead of new content after changes.

Note:
- 7.5.0 has the same issue. I am on Gnome 45.4 on Arch.
- 7.4.1 does **not** have the issue.

**Comment by pingej77** [comment]
Same issue on Fedora 39.  OnlyOffice 8.9.1.31 (flatpak), using OneDrive Fuse drive - file appears to save, but does not save.  Saving a copy locally and using cp or rsync seems to work as a workaround.  

I'm not sure if's helpful, but I found this https://www.onlyoffice.com/blog/2020/04/save-and-force-save-in-onlyoffice-never-lose-a-document which seemed promising as a workaround, but it appears to be for an older version of OnlyOffice -- I cannot find those "common settings" anywhere in my version and the saving settings under "advanced settings" no longer have that option.

**Comment by XDmitryK** [comment]
@pingej77 Hi
In your case I think the problem is related to issue #1503

**Comment by DUB1401** [comment]
I also have this problem on Fedora Silverblue 40.

OnlyOffice 8.0.1 (Flatpak)

**Comment by tanjera** [comment]
I've discovered/reproduced this in 8.0.1 on a few Linux installations. I did some testing to help diagnose the problem @XDmitryK and it only happens for me on a FUSE mount, mounting either Google Drive or a Samba share through the desktop environment (Nautilus or Thunar) or through an rclone mount. However, when I mount the same Samba share using cifs-util and mount.cifs in /etc/fstab, no problem!

I will be circumventing the problem by writing changes to the local drive and using a bidirectional file sync to push changes to the cloud drive.

**Comment by tanjera** [comment]
> I've discovered/reproduced this in 8.0.1 on a few Linux installations. I did some testing to help diagnose the problem @XDmitryK and it only happens for me on a FUSE mount

While this issue existed for me with FUSE mounts made with different programs (Nautilus and rclone), I've found documentation that it is a known limitation of the vfs-cache documented at https://rclone.org/commands/rclone_mount/ under "Limitations", specifically:

> Without the use of --vfs-cache-mode this can only write files sequentially, it can only seek when reading. This means that many applications won't work with their files on an rclone mount without --vfs-cache-mode writes or --vfs-cache-mode full. See the [VFS File Caching](https://rclone.org/commands/rclone_mount/#vfs-file-caching) section for more info. When using NFS mount on macOS, if you don't specify |--vfs-cache-mode| the mount point will be read-only.

Relevant to this, Nautilus uses its own implementation of gvfs https://wiki.gnome.org/Projects/gvfs ... not sure what its limitations are (?).

Mounting Google Drive via rclone w/ --vfs-cache-mode=full I tested and was successfully saving files directly to the FUSE drive with all programs working as expected. The file changes propagated successfully across to Google Drive and were readable in a browser on a different machine (so not read from cache!) Since the issue with saving looks like a problem with file caching on FUSE and the layers used to mount/access it... unless it has to do with the way OnlyOffice is accessing the filesystem (doubtful), I think the only thing in the scope for OnlyOffice, especially since it is cross-platform and used in so many environments:

- Ensure that file saves are actually saved; (e.g. check actual filesystem output for parity with intended filestream output perhaps cached in memory? ... or check file timestamp for DateTime::Now() to ensure file is updated ... or is there an error being returned from the FUSE filesystem that isn't being received/handled by OnlyOffice?)
- Warn the user if the save was not successful (e.g. "It appears the saved file was not written to the hard drive; Please <generic steps to take go here>")

In the meantime, @pingej77 @doryiii @kabe-inc (OS?) @dernippel and the people posting in #1503 may want to check if their filesystem mount has known caching limitations. Perhaps worth trying rclone w/ the vfs-cache-mode=writes flag

**Comment by kabe-inc** [comment]
@tanjera nice work! In response to your question, I'm using EndeavourOS + Gnome + Nautilus + Onedriver to handle the filesystem mount.

**Comment by alexrojo77** [comment]
@tanjera I wouldn't consider myself an expert at linux so not sure how to apply some of the mounting options you mention.  If you can can provide some instructions, I might be able to figure out how to do it, test it and report back.

As for the the caching limitations, somehow I don't think this is it even though I'm hardly an expert.  As mentioned in my original issue, I was able to manually install an older version (7.5.0.127) and was able to open and save files without any problems.  This is all using the same file browser and not making any changes to how the file server is mounted.  I also just installed the latest flatpak version (8.0.1.31) alongside the old version and I only experience the not saving issue with the newer version.  The mounting and way I access the file all being the same.  As doryiii said, a notification window does come up saying it is saving the document but when the file is re-opened, the new changes aren't there and the date/time stamp isn't updated either suggesting the file hasn't been saved.

**Comment by tanjera** [comment]
@alexrojo77 I agree with you- you've definitely seen a change in behavior between versions of OnlyOffice and ensuring that the program behaves as is best for the operating system might be room for this program to better itself. One of the reasons caching is needed (at least with VFS/rclone per the docs) are that if a file is being read and written to simultaneously, it will fail... I suspect handling the memory->file streams may be what changed in the versions. I tried to sift through the OnlyOffice code but couldn't find the save routine.

As far as mounting with a write cache enabled, here are the steps I used:
- Install rclone (your comment from November said you were on Debian- `sudo apt install rclone`
- Setup rclone using `rclone config` (does not need to be run as sudo/root)
  - Go through the steps of setting up a cloud drive and authenticating
- Mount the drive (you do need write access to the mounting folder!) using the `rclone mount --vfs-cache-mode=writes --daemon clouddrivename: /mount/directory/here

For example, the line in my crontab is `@reboot rclone mount --vfs-cache-mode=writes --daemon gdrive: "/mnt/Google Drive/"` as I named my cloud drive "gdrive" in `rclone config`

I've used OnlyOffice a bunch over the last few days without a single problem since setting the cache mode.

**Comment by XDmitryK** [comment]
@alexrojo77 
I can confirm - this is bug, so I have created issue #67855.
@tanjera Thanks for your feedback.
I think Google Drive is not working due to an issue #1442. We are in the process of solving this problem

**Comment by fnrfarid** [comment]
Hi Guys, is there any update on this issue? 

What's the best alternative/workaround?

**Comment by XDmitryK** [comment]
Unfortunately, there is no news yet about fixing these problems. Work in progress.

**Comment by alba4k** [comment]
Does anyone know if this is going to get a fix anytime in the near future?

Alternatively, does somebody know of a onedrive client that doesn't suffer of this? (I'm currently using onedriver) 
I just want something that can mount my onedrive just as if it was an external drive or smth

**Comment by pingej77** [comment]
> Does anyone know if this is going to get a fix anytime in the near future?
> 
> Alternatively, does somebody know of a onedrive client that doesn't suffer of this? (I'm currently using onedriver) I just want something that can mount my onedrive just as if it was an external drive or smth

I am now using this one:  https://github.com/abraunegg/onedrive
It works well, but the downside is that it stores all the files on the local machine instead of caching.  So it has good performance and works cleanly, but will take up a lot of space.  You can configure it to only sync certain folders though so if you do that you can limit the size it takes up and maybe do that for your documents and then pair it with a FUSE drive for other folders, as an idea.

**Comment by alba4k** [comment]
That sounds like more work than just saving as a copy in onlyoffice and then using mv to overwrite the original file 

and I have ~80GB on onedrive, so not caching anything is a pretty big deal

**Comment by pingej77** [comment]
More one-time setup I guess.  If you only have one folder of office files it could work pretty handily.  I stored 23GB out of my ~100GB locally.  Like I said, not a perfect solution, but I'm just throwing out suggestions here.

**Comment by pingej77** [comment]
If you or anyone else does go that route, here's my sync_list file (see documentation on that github page:

`  1   │ #OneDrive Sync List
   2   │ 
   3   │ #Exclude game saves and other useless data in Documents folder
   4   │ !Documents/temp*
   5   │ !/Documents/Custom Office Templates
   6   │ !/Documents/Horizon Zero Dawn
   7   │ !/Documents/MAXON
   8   │ !/Documents/My Games
   9   │ !/Documents/RCT3
  10   │ !/Documents/Program Data
  11   │ !/Documents/The Witcher 3
  12   │ !/Documents/Universe Sandbox
  13   │ !/Documents/Virtual Machines
  14   │ !/Documents/REAPER Media
  15   │ !/Documents/Aurora DSP
  16   │ 
  17   │ 
  18   │ #include the following folders
  19   │ /Documents/
  20   │ /Code
  21   │ /Music
  22   │ /Pictures
`

**Comment by pallebone** [comment]
I also have this issue with a samba share to a windows pc and the snap package. If I open a file and edit and save it the changes are lost, so i have to copy the file to a local location then edit it and then copy it back.

**Comment by guiguizada** [comment]
any viable solutions for this yet? we have the same issue with an SMB file server on 8.1.1.27 (flatpak)

**Comment by MAG3845** [comment]
Any update? For the moment it works with `--vfs-cache-mount writes`  using RClone with a OneDrive mounting, but not really, rclone takes like 5-10 minutes to detect and upload the file and his modifications, not a big deal if you work alone but a lot if you work with a Team, the problem doesn't append with others Office Suite, like Office ( on Windows ), WPS Office ( on Linux, it's crap btw ), and LibreOffice ( on Linux ), just you save your file and it instants of your Cloud Provider, so the problem don't come from rclone or vfs-caching or FUSE, it's from how OnlyOffice handles file and theirs saving.

**Comment by dernippel** [comment]
I switched today from the flatpak version 8.2.0 to the debian package (see [Installing ONLYOFFICE Desktop Editors to Debian, Ubuntu and derivatives](https://helpcenter.onlyoffice.com/de/installation/desktop-install-ubuntu.aspx)). This will install version 8.2.0.143 (try "~$ desktopeditors --version" in terminal after install). I little test showed that I'm now able to write to the SMB-Share again.

I just uninstalled the flatpak through the gui and installed the apt-version using the guide I linked. I did nothing else.

**Comment by moritztim** [comment]
This might be a separate issue, but this just happened to me with a local folder.

**Comment by lonestarr-007** [comment]
Any update on this issue?  I am experiencing the same thing with Fedora 40, Gnome, flatpak version 8.2.2, files mounted with Onedriver.  I'd be willing to try dernippel's solution if there is an rpm package.  Unfortunately most everything I work with is on OneDrive so this renders OnlyOffice useless to me until this is resolved, I will just have to either use LibreOffice (which I have confirmed does not have this same issue) or MS Office web version.

**Comment by yllekz** [comment]
Update please. This bug needs to be borderline #1 in terms of priority. It does not occur on macOS, only Linux it seems.

**Comment by birdfird** [comment]
I just had the same issue with editing a spreadsheet over a smb share. Ubuntu 24.04 & Snap install version 8.2.2

**Comment by stephanlachnit** [comment]
Just had to bug appear to me as well - Flatpak version 8.2.2, saving does not work. Even saving a copy sometimes doesn't work. It worked fine at first, but then it just broke and now I'm unable to save anything to my local disk anymore.

**Comment by cast-away** [comment]
Same here,version 8.2.2.22 (flatpak). 6+ hours of very intense work lost...

**Comment by stephanlachnit** [comment]
I noticed that for me saving stopped working when adding an SVG picture - once I removed it, saving worked fine again

**Comment by alba4k** [comment]
Not the case for me, saving to onedriver sometimes fails even when just writing some text in a .docx

**Comment by youpie** [comment]
This issue is still happening for me, fedora 41 gnome 47. Onlyoffice 8.3.1 flatpak. It happened on a SMB share

**Comment by savionlee** [comment]
Discovered recently on Fedora 41, KDE 6.3.3, onedriver, onlyoffice 8.3.2.19 flatpak

**Comment by moritztim** [comment]
HOW IS THIS STILL OPEN THIS IS A MAJOR PROBLEM

**Comment by lonestarr-007** [comment]
Agreed, I just confirmed the bug still exists.  At one point I opened a topic for this on the OnlyOffice forum support and was basically referred to paid support to get this issue prioritized.  That is unacceptable for a bug like this.  As @youpie stated, this bug can easily cause significant data loss, especially for those who may not be aware of it until they've performed days worth of work that goes unsaved.  This needs to be #1 priority, above new features, because this renders this app completely useless for those of us who do work in OneDrive or other FUSE mounted file systems

**Comment by moritztim** [comment]
Considering that I consistently run into this issue when using git makes it worse, because that means this happens specifically when I'm trying to prevent data loss

**Comment by littlegraycells** [comment]
> this bug can easily cause significant data loss

This is me. I just lost a day's worth of spreadsheet work with a ton of formulas and calculations that is going to take at least another day to re-create. Extremely frustrating. This has happened to me twice before but both times I though I had made some sort of a mistake but after finding this issue I am certain it wasn't me. For what it's worth, I wasn't relying on any kind of auto-save either. I specifically saved my work multiple times during the edits. 

Very disappointing that this still hasn't been fixed.

**Comment by littlegraycells** [comment]
I should mention that I have been able to reproduce it reliably. For me this bug only happens if I'm editing a file on the network. Local editing seems to be working fine and I wasn't able to reproduce the error. 

I think at the very least, there should be an error if the file was not saved properly before the app quits.

**Comment by littlegraycells** [comment]
@alexrojo77 I'd recommend changing the title to something like
> OnlyOffice Desktop Editors does not save files while working directly from a network share and quits without any errors.

**Comment by littlegraycells** [comment]
@alexrojo77 To more accurately reflect the issue (and the hope that it gets more attention), I'd recommend changing the title to something like

> OnlyOffice Desktop Editors does not save files while working directly from a network share and quits without any errors.

**Comment by moritztim** [comment]
Since it also happens in git repositories, I suspect the issue is in some common property of the two. Maybe that could be reflected if someone knows the relevant term

**Comment by cast-away** [comment]
> [@alexrojo77](https://github.com/alexrojo77) To more accurately reflect the issue (and the hope that it gets more attention), I'd recommend changing the title to something like
> 
> > OnlyOffice Desktop Editors does not save files while working directly from a network share and quits without any errors.

This was not my experience. I was saving locally.

**Comment by theblackhole** [comment]
> > this bug can easily cause significant data loss
> 
> @littlegraycells: This is me. I just lost a day's worth of spreadsheet work with a ton of formulas and calculations that is going to take at least another day to re-create. Extremely frustrating.

Same. I discovered that I lost all my spreadsheet and writing work from yesterday (and this was a long day) and I'm completely discouraged from having to redo all of it. I use Onedriver precisely to avoid losing any work, and the opposite happened when I used OnlyOffice to edit a file in my Onedriver mounted folder. 😓 

OnlyOffice absolutely needs to **display an error** or at least **keep the files in the [recover folder](https://www.onlyoffice.com/blog/2023/07/how-to-recover-documents)** for a few days, **even when it's saved correctly**.
The fact that it's deleted when you save and close the app ironically eliminates any chance of recovery. Especially since, in most cases, documents stored in this cache aren't going to fill up disk space, so keeping them for a few days shouldn't be a problem.

No matter how difficult it is to find the source of the problem, no matter how difficult it is to resolve: the first step should be to have a reliable recovery system that you can rely on.

I'm not asking for a continuous history of file changes like Jetbrains' code editors (although that would be ideal, as this has saved me many times), just to adapt the current recovery functionality to make it more reliable by not deleting cached files. This would cover any potential issues that may arise.

This would avoid the kind of significant frustration that has a serious impact on OnlyOffice's reputation. Indeed, after this bad experience, I'm no longer sure I'd recommend using OnlyOffice for professional use, even though I was planning to prepare arguments for replacing Microsoft 365 in the company.
Personally, I'm going to adapt by copying my files from now on before editing, but as an IT professional, this isn't an approach I would dare impose on teams.

**Comment by littlegraycells** [comment]
I've just moved on to Libre Office. Yes, the MS Office compatibility isn't as good but honestly, I'll take that any day over losing work.

**Comment by Pawel-9215** [comment]
This just happened to me on windows. I saved file, closed Only Office. After opening my document again 90% of it was gone.

**Comment by PascalRepond** [comment]
> HOW IS THIS STILL OPEN THIS IS A MAJOR PROBLEM

Yes, this is very much a vicious and blocking bug. How it has not been acknowledged or worked around, let alone fixed, while the company boasts of its new AI features is beyond me.

**Comment by fenris** [comment]
Still happened to me on v9.0.3.29. Do you have the latest update with a temporary workaround?

**Comment by ErikHeller** [comment]
Same problem here when using OnlyOffice and editing files on an SMB mounted server.
Saving a new file with save as works, but overwriting the file with save or save as doesn't update the state of the spreadsheet. The "last updated" property of the file also doesn't get updated according to Ubuntu's file manager.
Version:  9.0.0.172 (snap) on Ubuntu 24.04 LTS

**Comment by alexrojo77** [comment]
It's unfortunate that OnlyOffice has not fixed this nasty bug.  I think it should be higher up in on their rating scale but at the end of the day, they're really only hurting their brand and image by NOT fixing it as people run into this bug and lose valuable time/work.  (BTW, in the past this was acknowledged and assigned to someone but someone has changed this information.)

Anyways, I wanted to re-share how I was able to get around this bug although I mainly use LibreOffice now because this.  I also understand people's computer setups are different so perhaps this may or may not be a solution and it may or may not work.  If you're going to try it, definitely test it out first before working on a big document!

So on my system, I'm on Ubuntu 24 LTS and I had installed this program via the App Center.  Naturally when I upgraded to the latest version of OnlyOffice, it went thru the App Center.  After upgrading to 7.5.1.23, I experienced this not saving issue with my excel and word files (yes it was sort of a gut punch after experiencing it).  I also tried it from Flatpak and same thing happened.  However what I found was that if i downloaded the deb installation file and manually installed it, files were properly saved.  In fact, I just uninstalled an older v8 that I had manually installed months ago and put on the latest version (9.0.3.29) and files accessed via SMB share were saved properly to my synology NAS.  I didn't do anything else "special".  Although not the cleanest and ideal solution, it does work for me.  Anyways, hopefully this helps someone.

**Comment by moritztim** [comment]
@Rita-Bubnova seems to be responsible for managing this issue. They have changed the title and assigned @XDmitryK and removed them again. Perhaps this ping will bring some attention back to the issue.

**Comment by alba4k** [comment]
Let's hope it does, cause this is without any doubt a major issue that often peevents me from being able to use OnlyOffice

**Comment by maya-razdan** [comment]
Yes I might stop using it for exactly this reason.

**Comment by doryiii** [comment]
I feel like the new bug title does not give the right sense of the blast radius of this bug. THis affect much more than synology NAS users; it affects any gvfs-mounted filesystem, which is a lot more serious and widespread than just "synology NAS users".

**Comment by savionlee** [comment]
Maybe we need to submit a customer support ticket so it gets more visibility. 

https://helpdesk.onlyoffice.com/hc/en-us/requests/new

I made a ticket in May and this is what they sent

> Vigen (ONLYOFFICE)
> 
> May 12, 2025, 16:06 GMT+3
> 
> Hello!
>  
> Thanks for reaching out to us. 
>  
> Sorry to hear that such case happened to you. Our technical assistance team is already informed about it and if you're talking about Desktop Editors application, our technical agents are always checking your comments, so they can work on fixing the issues in the next updates. 
>  
> I hope that such issues don't happen anymore. In case of questions, you can always ask them in our Forum.
 
> Best regards,
> Vigen
> ONLYOFFICE Support Team

**Comment by BrunoG1987** [comment]
To me the bug is also presented when saving files stored on my main disk. This bug is BLOCKING, because you can't lose a day of work. This should be fixed ASAP!

**Comment by DUB1401** [comment]
@maxkadushkin is any news fot this BLOCKING bug? Oh who will be assigned for this?

**Comment by BrunoG1987** [comment]
I tried to reproduce the error mentioned in my post above, and I have found that corrupted images are to blame, as already pointed out by @stephanlachnit. Now, the issue is that I can clearly see that these SVG files are corrupted, but failing to save only because an embedded image is not parsed/displayed correctly is a little bit too much. Do you know which library OO uses for processing SVG files? I would like to look at it to try to suggest a fix...

**Comment by alba4k** [comment]
Different issue, then

I had OO fail to save even simple files with just "Hello World" in them

**Comment by bapr5** [comment]
Lost a day of work because of this bug :c fix pls i dont want to use libreoffice

**Comment by pcasalinho** [comment]
- test 1
  - Start OO
  - Create new Spreadsheet file
  - make changes, save to smb share #OK
- test 2
  - open file from smb share with Files, open with ONLYOFFICE (Ubuntu file manager)
  - make changes, `SAVE` to smb share #changes are not saved, file modified date is changed
 - test 3
   - open file from smb share with Files, open with ONLYOFFICE (Ubuntu file manager)
   - make changes, `SAVE AS` new file name #new file has all the changes
- extra test
   - create new file, make changes, save to smb share #OK
   - Open created file with libreoffice #says file is corrupted

This is a major problem, cannot work with smb shares??? Worst of all, it gives no error, says the changes are saved but are not!

**Comment by moritztim** [comment]
Again, it also affects git repositories (that is files in a local, non-network directory that is managed by git)

**Comment by Beo-Coder** [comment]
I just tried a few different things and found (as others have described) that this problem only occurs when OnlyOffice is installed via the Software Manager. When I install OnlyOffice manually via the .deb file or using the .AppImage file from GitHub, saving to an SMB share works.
I have not tested if this also works with OneDrive.
I am using Ubuntu 24.04.03 LTS with GNOME 46.
This is not ideal, of course, but it is still a much better solution than continuing to lose data.

**Comment by pcasalinho** [comment]
> manually via the .deb file

It has the exact same problem. `sudo apt install ./onlyoffice-desktopeditors_amd64.deb`

**Comment by alexrojo77** [comment]
> > manually via the .deb file
> 
> It has the exact same problem. `sudo apt install ./onlyoffice-desktopeditors_amd64.deb`

i'm not 100% sure but running it via the command line isn't the same as downloading it from the OO website.  i downloaded the deb file here (Just scroll down to the version):
https://www.onlyoffice.com/download-desktop.aspx?utm_source=helpcenter&utm_medium=article&utm_campaign=onlyoffice-desktop&utm_content=desktop-installation

after it downloads, manually install it.  good luck.

**Comment by moritztim** [comment]
> > > manually via the .deb file
> > 
> > It has the exact same problem. `sudo apt install ./onlyoffice-desktopeditors_amd64.deb`
> 
> i'm not 100% sure but running it via the command line isn't the same as downloading it from the OO website.  

Downloading from the website is a prerequisite to that command. The command just installs the downloaded deb file. 

> after it downloads, manually install it.  good luck.


What do you mean by manually installing it?

**Comment by alexrojo77** [comment]
> 
> > after it downloads, manually install it.  good luck.
> 
> What do you mean by manually installing it?

i right clicked on the deb file then selected "open with software install".  a box should come up where you can click on install  (below that it'll say something like local file).  what i did could be the same as typing it into the command line...i'm not sure since i just don't know linux well enough.

**Comment by mbeps** [comment]
Happened to me today too twice. This is a very serious issue and I might stop using OnlyOffice. It happened on Spreadsheets and Writer. I am using Fedora 42 and OnlyOffice is from Flathub.

**Comment by Madac1** [comment]
can confirm this exact behavior using onedriver sync tool for onedrive, onlyoffice just saves fine the first time and the following saves doesn't

**Comment by mbeps** [comment]
For me I am using Fedora Linux. I have InSync installed which constantly syncs my documents folder with my Google Drive. Haven't checked whether this is the cause.

**Comment by moolight-seashell** [comment]
same here i've lost a full day of works (flatpak wrtiing to a veracrytpt drive)

is this happen when lock file exists ?

**Comment by moritztim** [comment]
I reached out to the developers on their Discord server. Here's a quote from the response:
> The initial scenario is related to known bug. Unfortunately, we're still working on it. I have added your request to internal tracksystem.

**Comment by moolight-seashell** [comment]
this rather critical issue is two years old

any news ?

**Comment by moritztim** [comment]
> this rather critical issue is two years old
> 
> any news ?

Literally right above your comment, yes.

**Comment by lonestarr-007** [comment]
> > this rather critical issue is two years old
> > any news ?
> 
> Literally right above your comment, yes.

That's not really news though.  That's acknowledging they know about the bug, which they've been doing for over a year already, but not really giving any sort of update.  I'm starting to wonder if they're bothering to do anything about it at all, or if they're taking the Microsoft approach and focusing on new features and ignoring bugs for what they think is a small number of (non-paying) users.

**Comment by lonestarr-007** [comment]
Their support contact page asks whether you have a paid license.  Let's face it, they're not going to fix this bug for users that are not paying them.  If they were they'd have done so long ago.

**Comment by bryanwieg** [comment]
also happens on windows. a lock file is left as well, making the original file read only. and the changes are lost, not even a snapshot before it closed is left

**Comment by BrunoG1987** [comment]
> also happens on windows. a lock file is left as well, making the original file read only. and the changes are lost, not even a snapshot before it closed is left

Yes, under Linux if you shown hidden files you can see them. So, i partially solved my issue checking for their existence before opening any file.

**Comment by HeinzWuert** [comment]
This isn't a specific Linux bug, but a general one: there's no warning when saving, and no local copy is offered.

When editing a cloud-hosted document via Synology WebDav the OnlyOffice editor does not detect or warn the user when the connection to the document server is lost. The editor continues to accept input, but all unsynchronized edits are lost once the session is closed or refreshed.

Steps to reproduce:

Open a document stored in the cloud 

Start editing.

Disconnect the network (e.g. disable Wi-Fi) or stop the OnlyOffice DocumentServer process.

Continue typing for a few minutes — **the editor shows no warning.**

Reconnect or reload the page.

All changes made while disconnected are lost.

Expected behavior:

The editor should immediately detect the loss of connection.

It should display a clear “Connection lost / Offline mode” warning.

Local unsaved edits should be cached and recoverable once the connection is restored (similar to Google Docs or Collabora Online).

Actual behavior:

**No visible warning is shown.

The status indicator still shows “Saved” or no status change.

All edits made during disconnection are lost permanently.**

Environment:

OnlyOffice DocumentServer: 9.1.0.167 (x64 exe)

Integration: Synology NAS


No Reverse proxy

Network conditions: LAN and Wi-Fi drop simulated

Additional notes:
This behavior can cause severe data loss for users working with unstable connections. It would be highly useful if the editor implemented:

a clear offline detection mechanism (WebSocket/HTTP heartbeat),

a visible “offline” indicator,

and an autosave cache or recovery option for local unsynced changes.

💡 Suggested enhancement

Implement an offline detection system similar to:

navigator.onLine status in browsers

WebSocket ping/pong with a visible reconnect indicator

Local temporary file caching (so users can recover unsaved changes)

**Comment by cortegaunam** [comment]
Still present in 9.1.0, the last version working well with fuse mount points was 7.4.1. I always check the new releases to verify if this has been corrected and if not I downgrade to 7.4.1 again.

**Comment by freekvh** [comment]
I am also experiencing this issue (ONLYOFFICE Desktop Editors Community version 9.1.0.173 (deb), on NixOS), using Samba on a Synology NAS,  mounted via Gnome Files.

If I open an existing file and add something and save, it is simply not there when I open the file again.

I can create a file, save and and retrieve it, but can't add anything to it.

At least one existing Excel file will, after adding anything to it, be corrupt (it opens with a question about UTF8 and TSV separators, after accepting, the file is garbled and can't be opened in Excel anymore, LibreOffice tries repairing though and succeeds,  but repaired file does not have the additions). This is what opening the corrupted file looks like:

<img width="412" height="302" alt="Image" src="https://github.com/user-attachments/assets/23100d06-5add-4a97-a893-1c94cf1007d9" />

None of this is happening via when using the local filesystem.

EDIT

Some more info, I mounted via cifs-utils, and then indeed the corruption does not occur, and everything works normally (you may have read otherwise here previously, but that was my own fault, when correctly mounting as the current user, it works as expected, additions are saved, no corruption.)

**Comment by lheinke** [comment]
Same issue here, running Ubuntu 25.10 and OnlyOffice version 9.2.1.43 (x64 deb). There should be at least some indication/warning that this currently does not work in the program. It is not really acceptable for a professional office suite like ONLYOFFICE to cause loss of file changes like this.

**Comment by moolight-seashell** [comment]
still no news about this ?

**Comment by Zack-83** [comment]
Same problem here. I am on Linux Mint 22.3 - Cinnamon 64-bit with OnlyOffice Community-Version 9.2.1.43 (x64 flatpak)

**Comment by DrKsantor** [comment]
Issue reproduced in ipad air 2024, iOS 26.2.1 even for local files.

**Comment by Fabccc** [comment]
Same issue on a NFS server with ArchLinux 6.18.9 and onlyoffice Community-Version 9.3.0.140 (x64 deb)

**Comment by lonestarr-007** [comment]
Let's face it, OnlyOffice is not going to repair this bug.  They've acknowledged they know about it and have had 2 years to fix it.  I've switched to FreeOffice, it has equally as good compatibility with MS Office and does not have this bug.  I'm unsubscribing from this thread because I've accepted that OnlyOffice is not going to do anything about this problem.

**Comment by Zack-83** [comment]
Here is a workaround which worked for me (similar to what explained [above](https://github.com/ONLYOFFICE/DesktopEditors/issues/1436#issuecomment-2081779674)). Some folder managers like [Smb4k](https://howtoinstalls.com/ubuntu/smb4k.html#smb4k-screenshots) allow including remote folders like as if they were local, thus "tricking" OnlyOffice to save and synchronise documents.

**Comment by angussidney** [comment]
I ran into this issue today as well - lost several hours of work because OnlyOffice was indicating that the file was saved, even though it wasn't actually saved properly to disk. 

I understand that implementing a proper fix for this behaviour may be quite time consuming, however @tanjera's suggestion of adding a warning seems quite reasonable:

> * Ensure that file saves are actually saved; (e.g. check actual filesystem output for parity with intended filestream output perhaps cached in memory? ... or check file timestamp for DateTime::Now() to ensure file is updated ... or is there an error being returned from the FUSE filesystem that isn't being received/handled by OnlyOffice?)
> 
> * Warn the user if the save was not successful (e.g. "It appears the saved file was not written to the hard drive; Please ")

**Comment by user-mb** [comment]
I encountered files not saving on Linux in a Rclone mounted cloud storage.
No indication of failure by the editor.
Could it be that some timestamps are not correctly refreshed when saving and therefore changes are ignored by a network drive that doesn't hash everything?

**Comment by Snicelory** [comment]
HOW is this not fixed yet?!
I encountered this issue now too, on local files even. Haven't lost work really, but it baffles me, how this is still a thing.
I also can't get it fixed temporarily, so please, how can I get to work with OnlyOffice again, which I like a lot. As long as I don't encounter more bugs, that seem to linger without being touched at all....

**Comment by Snicelory** [comment]
Temporary fix:
Save the document with another document editor like LibreOffice or Collabora Office making any change. After that, OnlyOffice will save changes again.

OMG why does this still exist?! Isn't this open source...?
Please, if someone knows programming, take a look at this..

**Comment by JocelynDelalande** [comment]
Viable workaround (under linux) : uninstall flatpak version and reinstall via .deb version.

Worked for me at least, on two computers.

**Comment by user-mb** [comment]
> Viable workaround (under linux) : uninstall flatpak version and reinstall via .deb version.
> 
> Worked for me at least, on two computers.

I was using the AUR package based on the .deb and had the issue.
